language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/FailStatementResolver.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.codeaction;
+
+import io.ballerina.compiler.syntax.tree.DoStatementNode;
+import io.ballerina.compiler.syntax.tree.ForEachStatementNode;
+import io.ballerina.compiler.syntax.tree.LockStatementNode;
+import io.ballerina.compiler.syntax.tree.MatchStatementNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeTransformer;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.WhileStatementNode;
+
+/**
+ * This visitor is used to resolve the regular compound statement node of a given node.
+ * 
+ * @since 2.0.0
+ */
+public class FailStatementResolver extends NodeTransformer<Node> {
+    
+    public Node getRegComNSmtNode(Node node) {
+        return node.apply(this);
+    }
+
+    @Override
+    public Node transform(WhileStatementNode whileStatementNode) {
+        Node whileStatementConditionNode = whileStatementNode.condition();
+        if (whileStatementConditionNode.kind() == SyntaxKind.CHECK_EXPRESSION 
+                || whileStatementConditionNode.kind() == SyntaxKind.FAIL_STATEMENT) {
+            return whileStatementNode.whileKeyword();
+        }
```

Comment:
What is the use of this clause?

Reply:
line 40-43 suppose to handle giving no completions when we have check or fail expression inside a while statement's condition. Line 43 can be improved though.

Reply:
Without checking for the content within the expression, what if we consider the range where the particular action is triggered. In this case, we have a code action based on the diagnostics and before reaching this point we know that we are treating a given range and also this visitor is not a generic visitor. Hence IMHO we can consider the range. WDYT?

Reply:
+1